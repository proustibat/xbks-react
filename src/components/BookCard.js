import React from 'react';
import numeral from 'numeral';

export default class BookCard extends React.Component {
    constructor( props ) {
        super( props );

        this.state = {
            isbn: props.book ? props.book.isbn : '',
            title: props.book ? props.book.title : '',
            price: props.book ? ( props.book.price ).toString() : '',
            cover: props.book ? props.book.cover : '',
            synopsis: props.book ? props.book.synopsis : []
        }
    }

    render() {
        return (
            <div className="bookcard">
                <h1 className="bookcard__title">{ this.state.title }</h1>

                <div className="bookcard__row">
                    <div className="bookcard__cell-cover">
                        <img className="bookcard__cover" src={ this.state.cover } alt={ `cover for ${ this.state.title }` } />
                        <p className="bookcard__price">
                            <span>{ numeral( this.state.price ).format( '0.00$' ) }</span>
                        </p>
                    </div>
                    <div className="bookcard__cell-details">
                        <div className="bookcard__details">
                            <div className="bookcard__description">{ this.state.synopsis.join( ' ' ) }</div>
                        </div>
                    </div>
                </div>

            </div>
        );
    }
}

