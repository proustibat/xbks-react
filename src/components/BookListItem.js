import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const BookListItem = ( { isbn, title, price, synopsis } ) => (
    <Link to={ `/book/${ isbn }` } className="list-item">

        <div>
            <h3 className="list-item__title">{ title }</h3>
            <span className="list-item__sub-title">{ synopsis[ 0 ] }</span>
        </div>

        <h3 className="list-item__data">{ numeral( price ).format( '0.00$' ) }</h3>

    </Link>
);

export default BookListItem;
