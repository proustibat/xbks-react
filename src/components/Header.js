import React from 'react';
import Headroom from "react-headroom";
import { Link } from 'react-router-dom';

const Header = () => (
    <Headroom>
       <header className="header">
           <div className="content-container">
               <div className="header__content">
                   <Link className="header__title" to="/" >
                       <h1>XBKS</h1>
                   </Link>
                   <button className="button button--link">Cart</button>
               </div>
           </div>
       </header>
    </Headroom>
);

export default Header;
