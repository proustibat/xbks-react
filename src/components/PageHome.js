import React from 'react';

const PageHome = () => (
    <div>
        <h1>Hello XBKS</h1>
    </div>
);

export default PageHome;
