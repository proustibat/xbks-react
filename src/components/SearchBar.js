import React from 'react';
import { connect } from 'react-redux';
import { setSearchTerm } from "../actions/filters";

export class SearchBar extends React.Component {

    handleChange = e => {
        this.props.setSearchTerm( e.currentTarget.value );
    };

    render() {
        return (
            <div className="content-container">
                <input
                    className="searchbar__input"
                    type="text"
                    placeholder="Search"
                    value = { this.props.searchTerm }
                    onChange = { this.handleChange }
                />
            </div>
        );
    }
}

const mapStateToProps = state => ( {
    searchTerm: state.filters.searchTerm
} );

const mapDispatchToProps = dispatch => ( {
    setSearchTerm: text => dispatch( setSearchTerm( text ) )
} );

export default connect( mapStateToProps, mapDispatchToProps )( SearchBar );
