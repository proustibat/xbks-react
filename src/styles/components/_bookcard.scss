.bookcard {
  font-size: $font-size-medium;
  font-weight: 300;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  padding: $s-size;
  margin: $m-size 0;
}

.bookcard__row {
  display: flex;
  flex-direction: row;
  margin-top: $s-size;
  @media (max-width: $desktop-breakpoint - .01rem) {
    flex-direction: column-reverse;
  }
}

.bookcard__cell-cover {
  width: 30%;
}
.bookcard__cell-details {
  width: 70%;
}

@media (max-width: $desktop-breakpoint - .01rem) {
  .bookcard__cell-cover,
  .bookcard__cell-details {
    width: 100%;
    padding: 0;
  }
  .bookcard__cover {
    width: 70%;
  }
}

.bookcard__cover {
  width: 100%;
  display: block;
  margin: auto;
}

.bookcard__details {
  padding: 0 0 0 $s-size;
}

.bookcard__title {
  margin: 0;
  border: 1px solid #cacccd;
  line-height: 1;
  text-transform: uppercase;
  font-size: $font-size-large;
  padding: $s-size;
  background: $off-white;
}

.bookcard__description {
  font-size: $font-size-small;
  margin: 0;
  line-height: 1.5;

  @media (max-width: $desktop-breakpoint - .01rem) {
    margin-bottom: $s-size;
  }
}

.bookcard__price {
  padding: $s-size;
  border: 1px solid #cacccd;
  margin: $s-size 0;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background: $off-white;

  span {
    font-weight: 400;
  }
}


