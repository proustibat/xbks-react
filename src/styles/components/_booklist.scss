.booklist {
  margin: $l-size auto;
}
