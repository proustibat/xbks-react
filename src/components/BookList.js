import React from 'react';
import { connect } from 'react-redux';
import BookListItem from './BookListItem';
import getVisibleBooks from "../selectors/getVisibleBooks";

export class BookList extends React.Component {
    render() {
        return (
            <div className="booklist content-container">
                { this.props.books.length === 0 ? (
                    <span>No Books</span>
                ) : (
                    this.props.books.map( book => (
                        <BookListItem
                            key = { book.isbn }
                            { ...book }
                        />
                    ) )
                ) }
            </div>
        );
    }
}

const mapStateToProps = state => ( {
    books: getVisibleBooks( state.books, { searchTerm: state.filters.searchTerm } )
} );

export default connect( mapStateToProps )( BookList );
