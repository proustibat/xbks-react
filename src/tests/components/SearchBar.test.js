import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar } from '../../components/SearchBar';

// TODO simulate on change input

test( 'Should render SearchBar component correctly without a searchTerm content', () => {
    const wrapper = shallow( <SearchBar
        searchTerm = ''
    /> );
    expect( wrapper ).toMatchSnapshot();
} );

test( 'Should render SearchBar component correctly with a searchTerm content', () => {
    const wrapper = shallow( <SearchBar
        searchTerm = 'Hello my searchterm'
    /> );
    expect( wrapper ).toMatchSnapshot();
} );

test( 'Should handle text change', () => {
    const setSearchTerm = jest.fn();
    const value = 'new input value';

    const wrapper = shallow( <SearchBar
        setSearchTerm={ setSearchTerm }
    /> );

    wrapper.find( 'input' ).first().simulate( 'change', { currentTarget: { value } } );
    expect( setSearchTerm ).toHaveBeenLastCalledWith( value );
} );
