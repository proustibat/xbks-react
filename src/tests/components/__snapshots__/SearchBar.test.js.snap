// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SearchBar component correctly with a searchTerm content 1`] = `
<div
  className="content-container"
>
  <input
    className="searchbar__input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value="Hello my searchterm"
  />
</div>
`;

exports[`Should render SearchBar component correctly without a searchTerm content 1`] = `
<div
  className="content-container"
>
  <input
    className="searchbar__input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
